apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/reward_controller.ex

```diff
@@ -90,21 +81,76 @@ defmodule BlockScoutWeb.API.RPC.RewardController do
+
+      {:voter_format, :error} ->
+        render(conn, :error, error: "One or more voter addresses are invalid")
+
+      {:date_param, {:error, _}} ->
+        render(conn, :error, error: "Please only ISO 8601 formatted dates")
+    end
+  end
+
+  def getvalidatorsrewards(conn, params) do
+    with {:validator_address_param, {:ok, validator_address_param}} <- fetch_address(params, "validatorAddresses"),
+         {:validator_format, {:ok, validator_address_hashes}} <-
+           to_address_hashes(validator_address_param, "validatorAddresses"),
+         {:date_param, {:ok, from, _}} <- fetch_date(params["from"]),
+         {:date_param, {:ok, to, _}} <- fetch_date(params["to"]),
+         {:ok, rewards} <- ValidatorRewards.calculate_multiple_accounts(validator_address_hashes, from, to) do
+      render(conn, :getvalidatorsrewards, rewards: rewards)
+    else
+      {:validator_address_param, :error} ->
+        render(conn, :error, error: "Query parameter 'validatorAddresses' is required")
+
+      {:validator_format, :error} ->
+        render(conn, :error, error: "One or more validator addresses are invalid")
+
+      {:date_param, {:error, _}} ->
+        render(conn, :error, error: "Please only ISO 8601 formatted dates")
+    end
+  end
 
-      {:error, :not_found} ->
-        render(conn, :error, error: "Group address does not exist")
+  def getvalidatorgroupsrewards(conn, params) do
+    with {:group_address_param, {:ok, group_addresses_param}} <- fetch_address(params, "groupAddresses"),
+         {:group_format, {:ok, group_address_hashes}} <- to_address_hashes(group_addresses_param, "groupAddresses"),
+         {:date_param, {:ok, from, _}} <- fetch_date(params["from"]),
+         {:date_param, {:ok, to, _}} <- fetch_date(params["to"]),
+         {:ok, rewards} <- ValidatorGroupRewards.calculate_multiple_accounts(group_address_hashes, from, to) do
+      render(conn, :getvalidatorgroupsrewards, rewards: rewards)
+    else
+      {:group_address_param, :error} ->
+        render(conn, :error, error: "Query parameter 'groupAddresses' is required")
+
+      {:group_format, :error} ->
+        render(conn, :error, error: "One or more group addresses are invalid")
+
+      {:date_param, {:error, _}} ->
+        render(conn, :error, error: "Please only ISO 8601 formatted dates")
     end
   end
 
-  defp fetch_address(params, key) when key == "voterAddress" do
+  defp fetch_address(params, key) when key == "voterAddress" or key == "voterAddresses" do
     {:voter_address_param, Map.fetch(params, key)}
   end
 
-  defp fetch_address(params, key) when key == "groupAddress" do
+  defp fetch_address(params, key) when key == "groupAddress" or key == "groupAddresses" do
     {:group_address_param, Map.fetch(params, key)}
   end
 
-  defp fetch_address(params, key) when key == "validatorAddress" do
+  defp fetch_address(params, key) when key == "validatorAddress" or key == "validatorAddresses" do
```

Comment:
this feels like some bad design, we are seemingly handling `validatorAddress` and `validatorAddresses` differently, with different keys. It would make for a simpler api to just handle a list of addresses with the same key, where the list can contain 0..* addresses and the application can just deal with this.